[develop] Merge forward from 2018.3 to develop

rallytime opened the pull request:
Conflicts:
  - salt/netapi/rest_tornado/saltnado.py
  - tests/integration/states/test_file.py
  - tests/integration/states/test_git.py
  - tests/unit/modules/test_tls.py

terminalmage committed:
Make error more explicit when PKI dir not present for salt-call

twangboy committed:
Put some error checking in the shell command

Fix the example in the docs

twangboy committed:
Remove unnecessary try/except blocks

twangboy committed:
Remove redundant try/except block from state`

twangboy committed:
Fix build_env scripts

Uses `cmd /c` when launching commands to avoid the hang when installing
python 3
`clean_env.bat` will uninstall Py2 and Py3 in a single run

twangboy committed:
Remove erroneous pop statement

twangboy committed:
Fix winrm powershell script

Removes the `Hostname` paramater in the winrm create command. It is not
needed. It must match the hostname on the certificate, which is not
defined in the case of the Remote Desktop certificate
Removes extra spaces

terminalmage committed:
Force null bytes to be str types

When doing membership checks in a string, the string which is being
checked will be coerced to unicode if the null byte is a unicode type,
leading to a UnicodeDecodeError. This fixes that by forcing the null
byte literals to be str types. If the string on which the membership
check is being performed is a unicode type, then it will work just fine
since a null byte that is a str type will cleanly decode to a unicode
type even when the system encoding is ascii.

terminalmage committed:
Make salt.utils.stringutils.to_binary work for bytestrings

Also make sure we're testing bytestring equivalents to each of the
strings in the test.

terminalmage committed:
PY3: Ensure binary contents work with file.managed

terminalmage committed:
Use salt.utils.stringutils.is_binary to check if contents are binary

terminalmage committed:
file.managed: Add test to ensure binary contents work

mattp- committed:
GH-47000: add proper handling of full_return in cmd_subset

as noted in the original issue, full_return wasnt being handled properly. This
rectifies that.

terminalmage committed:
Fix calls to file.lsattr when lsattr is not installed

This was partially fixed before 2018.3.0, but some careless programming
since then has caused problems to return. This fixes those by making
`file.lsattr` return `None` when lsattr is not installed, and then fixes
all references to take this into account.

terminalmage committed:
Fix diffing binary files in file.get_diff

dwoz committed:
Fix fileserver roots tests

dwoz committed:
Jinja test fixes

dwoz committed:
Clean up cruft

Commit:
Merge pull request #47039 from twangboy/win_fix_winrm_script

Fix winrm powershell script

twangboy committed:
Cast vdata to appropriate type in reg state

Creates a `cast_vdata` function in the reg util
Calls that function from the state to put the data in the appropriate
type

Commit:
Lint: Remove unused import

dwoz committed:
fix py3 wart, encode os.linesep

gtmanfred committed:
use the recommended opennebula lookup method

Commit:
Catch an operation on a closed socket in a test

This failure wouldn't be related to the test in question so it should
be safe to ignore.

dwoz committed:
Use salt.utils.fopen

Commit:
Merge pull request #47055 from bloomberg/GH-47000

#47000 - add proper handling of full_return in cmd_subset

dwoz committed:
Fix py3 os.linesep wart

Commit:
Pass the timeout variable to the CLI when calling salt in tests

terminalmage committed:
Use decorators for temp files/dirs in test suite

This adds a new decorator which creates a temporary directory and cleans
it after the test completes. It also modifies an existing decorator for
creating temporary files so that it accepts arguments, which will be
passed through to salt.utils.files.mkstemp().

terminalmage committed:
Move back to using null byte check for contents

terminalmage committed:
Use the same way of defining contents in both file.managed states

dwoz committed:
Kitchn should ignore artifacts directory

gtmanfred committed:
Merge pull request #47074 from dwoz/ignore_artifacts

Kitchn should ignore artifacts directory

terminalmage committed:
pillar_ldap: Load config options as str types

Also make the same str type enforcement apply to all calls to _config()
in ldapmod.py.

Commit:
Merge pull request #47069 from cachedout/match_timeout_arg

Pass the timeout variable to the CLI when calling salt in tests

dwoz committed:
Merge pull request #47064 from dwoz/roots_tests_fix

Fix fileserver roots tests

dwoz committed:
Fix failing state test by normalizing line endings

Fixes:

integration.modules.test_state.StateModuleTest.test_issue_1896_file_append_source

dwoz committed:
Fix py3 wart (chr vs bytesstring)

gtmanfred committed:
Merge pull request #47067 from gtmanfred/2017.7

use the recommended opennebula lookup method

gtmanfred committed:
Merge pull request #47077 from dwoz/test_state_fix

Fix failing state test by normalizing line endings

gtmanfred committed:
dont allow using no_use_wheel for pip 10.0.0 or newer

The --no-use-wheel flag has been deprecated in pip 10.0.0

twangboy committed:
Merge branch '2017.7' into jinja_test_fix

twangboy committed:
Merge pull request #47065 from dwoz/jinja_test_fix

Jinja test fix

terminalmage committed:
yamlloader: Properly handle colons in inline dicts

PyYAML's scanner is not written to handle Python's unicode literal
strings, since they are not part of the YAML spec. With all strings
loaded in the YAML context now being unicode types, this causes errors
when scanning the document to construct the next node. The unicode
string literal is misidentified as a plain scalar and any colons within
it are interpreted as delimiting a mapping node. Since no whitespace appears
after the colon, this causes scanning error.

To fix it, we check for this scanning error and reset the scanner's
position and check if the first char is the "u" prefix. If so, and if
the following char is a quote, then we attempt to scan for a string
literal instead.

terminalmage committed:
Add additional test for plain scalars

terminalmage committed:
Fix path in log message

The actual error refers to the parent dir of the master/minion/etc. pki
subdir (i.e. /etc/salt/pki instead of /etc/salt/pki/minion). Therefore,
the startswith check needs to look for the parent of the pki_dir.

terminalmage committed:
Make error more specific to aid in troubleshooting

Commit:
Merge pull request #47068 from cachedout/catch_value_error_socket_test

Catch an operation on a closed socket in a test

twangboy committed:
Add integration tests for reg state

Fix issues with binary data found in testing

dwoz committed:
AsyncTestCase is required for AsyncEventPublisher

Other tests that use AsyncTestCase's io_loop will break if AsyncTestCase
is not subclassed when a test instantiates AsyncEventPublisher.

Commit:
Merge branch '2018.3' into issue46909

dwoz committed:
Merge pull request #47108 from dwoz/async_test_fix

Fix unit.utils.test_event.TestAsyncEventPublisher.test_event_subscription

kstreee committed:
Fixes misusing of the timeout option.

Commit:
fix pip.req import error in pip 10.0.0

gtmanfred committed:
fix other tests

Commit:
add ImportError

gtmanfred committed:
fix pip module for 10.0.0

Fixes #46971

Commit:
Merge branch '2017.7' into '2018.3'

Conflicts:
  - tests/integration/modules/test_state.py
  - tests/unit/templates/test_jinja.py
  - tests/unit/test_minion.py

Commit:
Update old utils paths references to use new paths

gtmanfred committed:
fix tests

gtmanfred committed:
Merge pull request #47037 from twangboy/fix_dev_scripts

Fix build_env scripts

terminalmage committed:
Remove unused imports

twangboy committed:
Fix tests for py3, enable tearDown

gtmanfred committed:
add __cli for master processes

terminalmage committed:
Rewrite flaky utf8 state to make it easier to troubleshoot

Commit:
lint: get_context is in stringutils.py now

terminalmage committed:
Special check specifically for bytes types

Commit:
Merge pull request #47102 from gtmanfred/2017.7

dont allow using no_use_wheel for pip 10.0.0 or newer

Commit:
Merge pull request #47129 from rallytime/bp-47121

Back-port #47121 to 2017.7

Ch3LL committed:
Add user integration tests for windows OS

Commit:
Merge pull request #47007 from twangboy/fix_46968

Fix some issues with the win_servermanager module

Commit:
Merge pull request #47131 from gtmanfred/cli

add __cli opts variable for master processes

terminalmage committed:
Fix incorrect logic in exception check

Commit:
Merge pull request #47041 from terminalmage/issue46909

Force null bytes to be str types

Commit:
Merge pull request #47134 from Ch3LL/user_win_test

Add user integration tests for windows OS

mcalmer committed:
provide kwargs to pkg_resource.parse_targets required to detect advisory type

mcalmer committed:
fix invalid string compare

gtmanfred committed:
Merge pull request #47107 from twangboy/fix_46932

Fix issues with reg state, add tests

Commit:
Merge pull request #40961 from terminalmage/issue40948

Make error more explicit when PKI dir not present for salt-call

Commit:
Merge pull request #47076 from terminalmage/issue46859

pillar_ldap: Load config options as str types

Commit:
Merge pull request #47104 from terminalmage/issue46929

yamlloader: Properly handle colons in inline dicts

Commit:
Merge pull request #47058 from terminalmage/lsattr

Fix calls to file.lsattr when lsattr is not installed

gtmanfred committed:
fix metadata grain for py3 and windows

Fixes #47168

Commit:
Merge pull request #47110 from kstreee/fix-misusing-of-timeout

Fixes misusing of the timeout option.

terminalmage committed:
mocked file_roots and pillar_roots should be dicts

Commit:
Merge pull request #47061 from terminalmage/issue47042

Fix diffing binary files in file.get_diff

Commit:
Updage jenkins module autodocs to use jenkinsmod name instead

Refs #46801

The jenkins module was renamed to jenkinsmod recently, but the autodoc
files were never updated.

We need to remove the jenkins.py docs from the docbuild server, but we
can't do that if the new docs for jenkinsmod.py aren't building yet.

This PR updates the autodoc files to point to the new file.

Commit:
Merge pull request #47155 from mcalmer/fix-patchinstall

Fix patchinstall for yumpkg

terminalmage committed:
Make sure a str type is passed to os.walk

With strings internally being unicode types in 2018.3.0, we end up
passing a unicode path to os.walk. Python 2 appears to have a bug in the
stdlib in some versions, in which posixpath can't handle unicode types.
This commit ensures that we pass a str type to os.walk in our os.walk
helper.

dwoz committed:
Allow non admin name based runs on windows

Commit:
Merge pull request #47070 from terminalmage/with_tempdir

Use decorators for temp files/dirs in test suite

Commit:
Merge pull request #47165 from terminalmage/issue47081

Make sure a str type is passed to os.walk

dwoz committed:
Use warnings to warn user

Ch3LL committed:
Add status module integration modules tests for Windows

Commit:
Test fix: remove tornado testing lib from class

This use in the test class causing a failure in this branch that doesn't occur
in the 2017.7 branch. This tornado lib is also not necessary for this particular
test class.

This might cause a failure in the windows tests, but I have already spoken
to @dwoz about this and he is going to address this in a separate PR if needed.

twangboy committed:
Add additional integration tests to whitelist

dwoz committed:
Merge pull request #47172 from dwoz/cover_without_admin

Allow non admin name based runs on windows

Commit:
Merge pull request #47185 from twangboy/add_tests

Add additional integration tests to whitelist

terminalmage committed:
salt-ssh: Do not attempt to match host/ip to minion ID if reverse lookup fails

When the reverse lookup fails, `salt.utils.network.ip_to_host()` returns
`None`. Later, when we attempt to look through the roster(s) for a minion ID
matching the target's host, a roster entry without a `host` key will
mean that `roster_data.get('host')` evaluates as None. This causes the
target expression to be expanded as `None`, when what we should do is
simply carry on without attempting to find a match in the roster(s).
This commit does just that.

Commit:
Merge pull request #47162 from terminalmage/bp-47161

Partial backport of #47161 to 2018.3 branch

Commit:
Merge pull request #47163 from rallytime/jenkins-autodoc

Updage jenkins module autodocs to use jenkinsmod name instead

Commit:
Merge pull request #47184 from Ch3LL/status_test

Add status module integration modules tests for Windows

terminalmage committed:
Add debug logging when ip_to_host fails

Commit:
Merge pull request #47122 from rallytime/merge-2018.3

[2018.3] Merge forward from 2017.7 to 2018.3

Commit:
Merge branch '2017.7' into '2018.3'

Conflicts:
  - salt/config/__init__.py
  - salt/minion.py
  - salt/modules/pip.py
  - salt/states/pip_state.py
  - salt/states/virtualenv_mod.py
  - tests/integration/modules/test_status.py

Commit:
Update old utils paths to new utils paths

mirceaulinic committed:
Fix #46918

Commit:
Lint: use full path for event utils function

Commit:
Merge pull request #47191 from terminalmage/issue47150

salt-ssh: Do not attempt to match host/ip to minion ID if reverse lookup fails

Commit:
Merge pull request #47202 from cloudflare/fix-46918

Fix #46918: add the TTL field

Commit:
Test fix: file strings must be unicode in master config

Commit:
Merge pull request #47168 from gtmanfred/metadata

fix metadata grain for py3 and windows

Commit:
Merge pull request #47195 from rallytime/merge-2018.3

[2018.3] Merge forward from 2017.7 to 2018.3

Commit:
Merge branch '2018.3' into 'develop'

Conflicts:
  - salt/netapi/rest_tornado/saltnado.py
  - tests/integration/states/test_file.py
  - tests/integration/states/test_git.py
  - tests/unit/modules/test_tls.py

terminalmage reviewed (approved):
Also checked out your `merge-develop` branch and made sure there are no new-to-develop uses of the `with_tempfile` test decorator, since they would now require parenthesis.

Commit:
Lint: Fix problems caused by bad merge conflict resolution

Commit:
Remove tests that were rewritten

The file.patch integration tests were rewritten in the develop branch
in PR #47010

They came back in during the merge-forward and should have been removed
again during the merge conflict resolution.
